fix: mypy hataları ile ilgili düzeltmeler

app/services/oauth_service.py:111: error: Argument 2 to "get_oauth_account" has incompatible type "Any | None"; expected "str"  [arg-type]
app/services/oauth_service.py:111: error: Argument 3 to "get_oauth_account" has incompatible type "Any | None"; expected "str"  [arg-type]
app/routers/auth.py:84: error: Incompatible types in assignment (expression has type "RedirectResponse", variable has type "JSONResponse")  [assignment]
app/main.py:12: error: Incompatible types in assignment (expression has type "FastAPI", variable has type Module)  [assignment]
app/main.py:19: error: Module has no attribute "add_middleware"  [attr-defined]
app/main.py:26: error: Module has no attribute "add_middleware"  [attr-defined]
app/main.py:36: error: Module has no attribute "include_router"  [attr-defined]
app/main.py:39: error: Module has no attribute "get"  [attr-defined]

--- a/backend/app/main.py
+++ b/backend/app/main.py
@@ -3,7 +3,7 @@ from fastapi.middleware.cors import CORSMiddleware
 from sqlalchemy import text
 from sqlalchemy.ext.asyncio import AsyncSession
 from app.config import settings
-import app.models  # noqa: F401 - ensure all SQLAlchemy models are registered
+import app.models as _app_models  # noqa: F401 - ensure all SQLAlchemy models are registered
 from app.database import get_db
 from app.routers import auth
 from starlette.middleware.sessions import SessionMiddleware

--- a/backend/app/routers/auth.py
+++ b/backend/app/routers/auth.py
@@ -1,6 +1,7 @@
 import logging
 import secrets
 import uuid
+from typing import Union
 from datetime import datetime, timezone
 
 import httpx
@@ -78,6 +79,7 @@ async def google_login(request: Request):
             f"&prompt=consent"
         )
         logger.info("Generated Google OAuth URL")
+        response: Union[JSONResponse, RedirectResponse]
         if request.query_params.get("format") == "json":
             response = JSONResponse(content={"login_url": auth_url})
         else:

--- a/backend/app/services/oauth_service.py
+++ b/backend/app/services/oauth_service.py
@@ -99,6 +99,9 @@ async def merge_oauth_accounts(
     provider_user_id = payload.get("provider_user_id")
     provider_email = payload.get("provider_email")
 
+    if not isinstance(new_provider, str) or not isinstance(provider_user_id, str):
+        raise ValueError("Geçersiz merge token: eksik provider bilgisi")
+
     result = await db.execute(
         select(User)
         .where(User.id == uuid_mod.UUID(user_id))

